test/unit/test-resources.js

```diff
@@ -45,6 +45,62 @@ describes.realWin('Resources', {amp: true}, (env) => {
     resources.pass_.cancel();
   });
 
+  function createResource(opts) {
+    const {
+      id,
+      isBuilt,
+      isBuilding,
+      state,
+      isDisplayed,
+      isFixed,
+      isInViewport,
+      prerenderAllowed,
+      renderOutsideViewport,
+      idleRenderOutsideViewport,
+      layoutPriority,
+      taskId,
+    } = {
+      id: '1',
+      isBuilt: false,
+      isBuilding: false,
+      state: ResourceState.NOT_BUILT,
+      isDisplayed: true,
+      isFixed: false,
+      isInViewport: true,
+      prerenderAllowed: false,
+      renderOutsideViewport: false,
+      idleRenderOutsideViewport: false,
+      layoutPriority: LayoutPriority.CONTENT,
+      taskId: 'resource#P',
+      ...opts,
+    };
+
+    const element = document.createElement('amp-el');
+    element.V2 = () => false;
+    element.isBuilt = () => isBuilt;
+    element.isBuilding = () => isBuilding;
+
+    const resource = new Resource(id, element, resources);
```

Comment:
nit: (not sure if this is better, but def more concise)
```js
Object.entries(opts, ([prop, val]) => {
  env.sandbox.stub(resources, prop).returns(val);
});
```

Reply:
Unfortunately, there are some minor naming differences. So I think it's ok for now to stub separately.